File: education/event_management/web_form/event_registration/event_registration.py
import frappe
from frappe import _

def get_context(context):
	# do your magic here
	pass
@frappe.whitelist(allow_guest=True)
def get_upcoming_events():
    return frappe.get_all(
        "Events",
        filters={"docstatus": 1},
        fields=["name", "event_title", "start_date", "end_date", "room", "college"],
        order_by="start_date asc"
    )
@frappe.whitelist(allow_guest=True)
def get_user_full_name(email):
    user = frappe.get_all(
        "User",
        filters={"email": email},
        fields=["full_name"],
        limit_page_length=1
    )
    if user:
        return user[0].full_name
    return ""

@frappe.whitelist(allow_guest=True)
def check_already_registered(event, faculty_email):
    """
    Check if email already registered for the event
    """

    if not event or not faculty_email:
        return {"status": "error", "message": "Missing event or email"}

    exists = frappe.db.exists(
        "Event Registration",
        {
            "event": event,
            "faculty_email": faculty_email,
            "docstatus": ["!=", 2]  # ignore cancelled
        }
    )

    if exists:
        return {
            "status": "duplicate",
            "message": f"{faculty_email} is already registered for this event."
        }

    return {"status": "ok"}

def after_insert(doc, method=None):
    """
    Auto-submit Event Registration after insert and check duplicate email
    """
    if not doc.event:
        frappe.throw(_("Please select an Event before submitting."))

    # Fetch the Event document
    event_doc = frappe.get_doc("Events", doc.event)
    already_registered = frappe.db.exists(
        "Event Registration",
        {
            "event": doc.event,
            "faculty_email": doc.faculty_email,
            "docstatus": ["!=", 2]
        }
    )

    # # If exists and not this same document
    # if already_registered and already_registered != doc.name:
    #     frappe.throw(
    #         _("{0} is already registered for this event.").format(doc.faculty_email)
    #     )

    # # Check if the email is already registered
    # already_registered = any(
    #     row.faculty_email == doc.faculty_email
    #     for row in getattr(event_doc, "faculty_register", [])
    # )

    # if already_registered:
    #     frappe.throw(_("{0} is already registered for this event.").format(doc.faculty_email))

    # Add the user to the Event's child table
    event_doc.append("faculty_register", {
        "faculty_email": doc.faculty_email,
        "faculty_name": doc.faculty_name
    })

    # Save Event doc
    event_doc.save(ignore_permissions=True)
    frappe.db.commit()

    # Auto-submit the Event Registration if still draft
    if doc.docstatus == 0:
        doc.submit()

    frappe.msgprint(_("✅ {0} has successfully registered to the event.").format(doc.faculty_name))

File: education/event_management/web_form/event_registration/event_registration.js
frappe.ready(function () {

	// Read event from URL query string
	const params = new URLSearchParams(window.location.search);
	const route_options = params.get("route_options");
	console.log("route_option:", route_options)

	if (route_options) {
		const opts = JSON.parse(decodeURIComponent(route_options));
		frappe.web_form.set_value("event", opts.event);
	}
	frappe.web_form.on('faculty_email', function (value) {
		if (!value) {
			frappe.web_form.set_value('faculty_name', '');
			return;
		}
		const email = value.doc.faculty_email;
		console.log(email)
		// Call server-side whitelisted method
		frappe.call({
			method: "education.event_management.web_form.event_registration.event_registration.get_user_full_name",
			args: { email: email },
			callback: function (r) {
				frappe.web_form.set_value('faculty_name', r.message || '');
			}
		});
	});

	let is_submitting = false;

	$('form button[type="submit"]').on('click', function (e) {

		if (is_submitting) return;

		e.preventDefault();

		is_submitting = true;

		let event = frappe.web_form.get_value("event");
		let email = frappe.web_form.get_value("faculty_email");
		let name = frappe.web_form.get_value("faculty_name");

		frappe.call({
			method: "education.event_management.web_form.event_registration.event_registration.check_already_registered",
			args: {
				event: event,
				faculty_email: email,
				faculty_name: name
			}
		}).then(r => {

			if (r.message && r.message.status === "duplicate") {
				// frappe.msgprint(" " + r.message.message);
				is_submitting = false;
			} else {
				$('form').submit();
				setTimeout(function () {
					window.location.href = '/events';
				}, 1000);
			}

		});

	});
});
